### ISSUE TITLE
Display trait implementation uses debug formatter

### ISSUE THREAD

**Issue by robfuscator** [polyfloyd/rust-id3#40]
I'm quite new to Rust, so I'm not sure if this is an actual issue - you can close it if you think it's not. Anyways, here are my thoughts:

I'm writing a library that abstracts every possible Error in a custom Error enum. When I try to read a tag from a file that does not exist, I receive an `id3::Error` that wraps around an `io::Error`. If, in my library, I format this Error using `write!(f, "{}", err);`, I get a String that looks like this:

`Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })`

If I format the same `io::Error` that is not wrapped up in your `id3::Error`, the String looks like this:

`No such file or directory (os error 2)`

I think this is due to your implementation of the `Display` trait, where you are using the debug formatter (`"{:?}"`):

```rust
impl fmt::Display for Error {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{:?}", self)
    }
}
```

I'm not sure if it is intended like this, or even best practice (as mentioned, I'm new to Rust). But as a user of your library I think it would be more convenient if I get the cleaner message using the default formatter (and maybe the 'uncleaner' message using the debug formatter).

Hope this makes sense.

--- edit:

I'd love to help on this one, if it is one :smile: .

**Comment by polyfloyd** [comment]
Hi!

The reason for using `fmt::Debug` in the implementation of `fmt::Display` is purely because it was easy to write. So yes, using `fmt::Display` for embedded errors would be neater ;)

If you are interested in improving this yourself, I would be happy to accept the patch. It is a bit more work than just copying the implementation of `fmt::Debug` because `ErrorKind` does not implement `fmt::Display`. So either implement `fmt::Display` for `ErrorKind` or add some special cases to the implementation for `Error`

**Comment by robfuscator** [comment]
Thanks for your answer and the hints. I'll do my best.

**Comment by polyfloyd** [comment]
Fixed in #41
